src/HRayData.hs

```diff
@@ -0,0 +1,131 @@
+{-# LANGUAGE AllowAmbiguousTypes       #-}
+{-# LANGUAGE ExistentialQuantification #-}
+{-# LANGUAGE RecordWildCards           #-}
+{-# LANGUAGE ScopedTypeVariables       #-}
+
+module HRayData
+    ( AABB
+    , AnyHit
+    , BoundingVolume
+    , ClosestHit
+    , Color
+    , Context
+    , D2
+    , D3
+    , Intersect
+    , Intersection
+    , Ray
+    , RayResult
+    , Scene
+    , SceneObject
+    -- | Functions
+    , getDouble
+    , getVector2
+    , getVector3
+    , raycast
+    ) where
+
+import Data.Map (Map, (!))
+import Linear.V2 (V2)
+import Linear.V3 (V3)
+
+import qualified Data.Map as Map hiding (Map)
+import qualified Linear.V2 as V2 hiding (V2)
+import qualified Linear.V3 as V3 hiding (V3)
+
+newtype D2 =
+    D2 (V2 Double)
```

Comment:
Formatting is really strange here... This is the reason I hate `hindent`. For example, this data type can be written in one line. There's no actual reason to put it into multiple lines, its declaration is not that long.